electron/main.ts

```diff
@@ -374,65 +378,97 @@ ipcMain.on('callTool', async (event, { tool, wine, prefix, exe }: Tools) => {
     command = `WINEPREFIX='${winePrefix}' ${wineBin} '${exe}'`
   }
 
-  console.log({ command })
+  Logger.info({message: `Trying to run command ${command}`, service: 'ipcMain::callTool::executingCommand'})
   return await execAsync(command)
 })
 
 /// IPC handlers begin here.
 
-ipcMain.handle('checkGameUpdates', () => LegendaryLibrary.get().listUpdateableGames())
+ipcMain.handle('checkGameUpdates', async () => {
+  await Logger.info({message: 'Checking all apps for updates', service: 'ipcMain::checkGameUpdates'})
+  return LegendaryLibrary.get().listUpdateableGames()
+})
 
 // Not ready to be used safely yet.
-ipcMain.handle('updateAll', () => LegendaryLibrary.get().updateAllGames())
+ipcMain.handle('updateAll', async () => {
+  await Logger.info({message: 'Updating all apps', service:'Game::updateAllGames'})
+  return LegendaryLibrary.get().updateAllGames()
+})
 
-ipcMain.handle('checkVersion', () => checkForUpdates())
+ipcMain.handle('checkVersion', async () => {
+  await Logger.info({message: 'Checking for a newer heroic version', service: 'ipcMain::checkVersion'})
+  return checkForUpdates()
+})
 
-ipcMain.handle('getMaxCpus', () => cpus().length)
-ipcMain.handle('getPlatform', () => process.platform)
+ipcMain.handle('getMaxCpus', () => {
+  return cpus().length
+})
+ipcMain.handle('getPlatform', () => {
+  return process.platform
+})
 
-ipcMain.on('createNewWindow', (e, url) =>
-  new BrowserWindow({ height: 700, width: 1200 }).loadURL(url)
-)
+ipcMain.on('createNewWindow', (e, url) =>{
+  return new BrowserWindow({ height: 700, width: 1200 }).loadURL(url)
+})
 
-ipcMain.handle('getGameInfo', async (event, game) => {
-  const obj = Game.get(game)
+ipcMain.handle('getGameInfo', async (event, appName) => {
+  await Logger.info({message: `Getting app info for ${appName} `, service: 'ipcMain::getGameInfo'})
```

Comment:
This await being called for +100 or + 200 games, will make the app a lot slower, I'm sure of that.

Reply:
This is only ran when a user goes into the game's page, not when they're in the library.